monitoring/src/main/java/org/polypheny/db/monitoring/core/MonitoringQueueImpl.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2019-2021 The Polypheny Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.db.monitoring.core;
+
+import com.google.common.collect.Sets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.polypheny.db.config.RuntimeConfig;
+import org.polypheny.db.monitoring.events.MonitoringEvent;
+import org.polypheny.db.monitoring.persistence.MonitoringRepository;
+import org.polypheny.db.util.background.BackgroundTask;
+import org.polypheny.db.util.background.BackgroundTask.TaskSchedulingType;
+import org.polypheny.db.util.background.BackgroundTaskManager;
+
+/**
+ * MonitoringQueue implementation which stores the monitoring jobs in a
+ * concurrentQueue and will process them with a background worker task.
+ */
+@Slf4j
+public class MonitoringQueueImpl implements MonitoringQueue {
+
+    // region private fields
```

Comment:
Why did you add these code arrangement comments here?

Reply:
Thats a feature of IntelliJ to nicely show structures of classes. We can remove them again if you find it disturbing:
![image](https://user-images.githubusercontent.com/40957033/127331426-8e145b1f-f8a7-411a-8069-cab7662bacb8.png)

Reply:
Of course, if you find them useful, we can put them in the code. For such small classes I don't really see the benefit, but for me it's fine to have the comments in the code if you find them useful.